drivers/include/periph/pm.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_pm Power Management
+ * @ingroup     drivers_periph
+ * @brief       The kernels power management interface
+ * @{
+ *
+ * This simple power management interface is based on the following assumptions:
+ *
+ * - CPUs define up to 4 power modes (from zero, the lowest power mode, to
+ *   PM_NUM_MODES-1, the highest)
+ * - there is an implicit extra idle mode (which has the number PM_NUM_MODES)
+ * - individual power modes can be blocked/unblocked, e.g., by peripherals
+ * - if a mode is blocked, so are implicitly all lower modes
```

Comment:
It seems that the implementation doesn't follow this rule. It selects the first unblocked power mode starting from the lowest mode [drivers/periph_common/pm.c](https://github.com/kaspar030/RIOT/blob/bb2fb6d28c3efda8db3d3582233be06034106f2c/drivers/periph_common/pm.c#L31)

``` c
void pm_set_lowest(void)
{
    pm_blocker_t blocker = (pm_blocker_t) pm_blocker;
    unsigned mode = 0;
    while (mode < PM_NUM_MODES) {
        if (! (blocker.val_u8[mode])) {
            break;
        }
        mode++;
    }
   ...
}
```